include/ck_tile/core/arch/amd_buffer_addressing.hpp

```diff
@@ -2666,6 +2712,17 @@ CK_TILE_DEVICE void amd_buffer_atomic_add(const thread_buffer<T, N>& src_thread_
                                           const bool dst_thread_element_valid,
                                           const index_t dst_element_space_size)
 {
+#if defined(__gfx11__)
+    if constexpr(std::is_same<T, fp16_t>::value)
+    {
+        // gfx11 does not support buffer atomic add in fp16 format.
+        // Thus, call the fallback method instead
+        ignore = dst_thread_element_valid;
+        ignore = dst_element_space_size;
+        atomic_add_impl<T, N>(&p_dst_wave[dst_thread_element_offset], src_thread_data);
+        return;
+    }
+#endif
```

Comment:
IMHO I think it was better to handle this in `buffer_view` file as this one is solely devoted to amd buffer instruction implementations.